src/dbcp/extract/acp_projects.py

```diff
@@ -1,10 +1,41 @@
 """Extract ACP projects from GCS archive."""
 
+import re
 from pathlib import Path
 
 import pandas as pd
 
 import dbcp
+import dbcp.extract
+import dbcp.extract.helpers
+
+
+def _extract_acp_projects_snapshots() -> pd.DataFrame:
+    """Extract a dataframe with all ACP quarters concatenated together.
+
+    Used for creating the ACP changelog.
+    """
+    file_paths = dbcp.extract.helpers.cache_gcs_archive_bucket_contents_locally(
+        gcs_dir_name="acp"
+    )
+    concat_df = pd.DataFrame()
+    quarter_to_month = {"1": 1, "2": 4, "3": 7, "4": 10}
+    for path in file_paths:
+        snapshot_df = pd.read_csv(path)
+        filename = path.parts[-1].split(".")[0]
+        match = re.match(r"projects_Q([1-4])_(\d{4})", filename)
+        if not match or match.lastindex < 2:
+            raise ValueError(
+                f"ACP filename '{filename}' is not formatted as expected (should match 'projects_Q{{1-4}}_{{YYYY}}'). "
```

Comment:
If we move to monthly snapshots, should we then also accept both quaterly naming conventions and monthly, and only transform to monthly if it's the former?

Reply:
Converted to a naming format with a full date in the file name `acp_projects_{YYYY}_{MM}_{DD}`